front/app/containers/Admin/settings/pages/PageEditor.tsx

```diff
@@ -103,173 +101,78 @@ const EditionForm = styled.div`
-      };
-    } else {
-      initialValues = {
-        title_multiloc: {
-          en: slug,
-        },
-        body_multiloc: {},
-      };
-    }
-
-    if (!isNilOrError(remotePageFiles)) {
-      initialValues['local_page_files'] = remotePageFiles;
-    } else {
-      initialValues['local_page_files'] = [];
-    }
-
-    return initialValues;
-  };
-
-  getFilesToAddPromises = (values: FormValues) => {
-    const { local_page_files } = values;
-    const localPageFiles = [...local_page_files];
-    const { page, remotePageFiles } = this.props;
-    const pageId = !isNilOrError(page) ? page.id : null;
-    let filesToAdd = localPageFiles;
-    let filesToAddPromises: Promise<any>[] = [];
-
-    if (!isNilOrError(localPageFiles) && Array.isArray(remotePageFiles)) {
-      // localPageFiles = local state of files
-      // This means those previously uploaded + files that have been added/removed
-      // remotePageFiles = last saved state of files (remote)
-
-      filesToAdd = localPageFiles.filter((localPageFile) => {
-        return !remotePageFiles.some((remotePageFile) =>
-          remotePageFile
-            ? remotePageFile.filename === localPageFile.filename
-            : true
-        );
-      });
-    }
-
-    if (pageId && !isNilOrError(filesToAdd) && filesToAdd.length > 0) {
-      filesToAddPromises = filesToAdd.map((fileToAdd: any) =>
-        addPageFile(pageId as string, fileToAdd.base64, fileToAdd.name)
-      );
-    }
-
-    return filesToAddPromises;
+  const initialValues = (
+    page: IPageData,
+    remotePageFiles: useRemoteFilesOutput
+  ) => {
+    return {
+      title_multiloc: page.attributes.title_multiloc,
+      body_multiloc: page.attributes.body_multiloc,
+      slug: page.attributes.slug,
+      local_page_files: remotePageFiles,
+    };
   };
 
-  getFilesToRemovePromises = (values: FormValues) => {
-    const { local_page_files } = values;
-    const localPageFiles = [...local_page_files];
-    const { page, remotePageFiles } = this.props;
-    const pageId = !isNilOrError(page) ? page.id : null;
-    let filesToRemove = remotePageFiles;
-    let filesToRemovePromises: Promise<any>[] = [];
-
-    if (!isNilOrError(localPageFiles) && Array.isArray(remotePageFiles)) {
-      // localPageFiles = local state of files
-      // This means those previously uploaded + files that have been added/removed
-      // remotePageFiles = last saved state of files (remote)
-
-      filesToRemove = remotePageFiles.filter((remotePageFile) => {
-        return !localPageFiles.some((localPageFile) =>
-          remotePageFile
-            ? localPageFile.filename === remotePageFile.filename
-            : true
-        );
-      });
-    }
-
-    if (pageId && !isNilOrError(filesToRemove) && filesToRemove.length > 0) {
-      filesToRemovePromises = filesToRemove.map((fileToRemove: any) =>
-        deletePageFile(pageId as string, fileToRemove.id)
-      );
-    }
+  const handleSubmit = (
```

Comment:
Then I would have to do the isNilOrError check in multiple places and I try to avoid that.

Reply:
Why? The `isNilOrError` check is in the nested function so it would still be only in one place right

Reply:
I need it before returning the component (for page.id) and also in two functions. If I pass it with the functions I only need to do the !isNilOrError(page) in 1 spot vs. 3 spots otherwise :-)